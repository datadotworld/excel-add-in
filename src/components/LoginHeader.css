/*
 * Copyright 2017 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
.login-header {
  background-color: #335c8c;
  height: 50px;
  color: #fff;
  display: flex;
  align-items: center;
  padding: 5px 10px;
}

.login-header .logo-full {
  height: 40px;
}

.login-header .dropdown {
  margin-left: auto;
}

.login-header .dropdown.btn-group button {
  border: none;
  box-shadow: none;
  padding: 0;
  line-height: 1;
  height: auto;
  font-size: 16px;
  color: #fff !important;
  background-image: none;
}

.login-header .dropdown-menu {
  z-index: 1060;
}