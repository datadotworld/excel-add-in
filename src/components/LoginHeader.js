/*
 * Copyright 2017 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  DropdownButton,
  MenuItem
} from 'react-bootstrap';

import dwLogo from '../static/img/dw-logo.png';

import './LoginHeader.css';
import analytics from '../analytics';

class LoginHeader extends Component {

  static propTypes = {
    user: PropTypes.object,
    logout: PropTypes.func
  }

  userMenuChanged = () => {
    analytics.track('exceladdin.header.logout.click');
    this.props.logout();
  }

  render () {
    const { user, page } = this.props;

    return (
      <div className={`login-header ${page}`}>
        <img src={dwLogo} alt="data world logo" className="dw-logo" />
        {user && <DropdownButton title={`@${user.id}`} pullRight bsSize='small' onSelect={this.userMenuChanged} id='dropdown-user'>
          <MenuItem eventKey='signout'>Sign out</MenuItem>
        </DropdownButton>}
      </div>
    );
  }
}

export default LoginHeader;
