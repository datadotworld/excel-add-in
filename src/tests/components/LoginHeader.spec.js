/*
 * Copyright 2017 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import LoginHeader from '../../components/LoginHeader';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<LoginHeader />, div);
});

it('renders login header - no user', () => {
  expect(renderer.create(<LoginHeader />).toJSON()).toMatchSnapshot();
});

it('renders login header - with user', () => {
  const user = { id: 'test' };
  expect(
    renderer.create(<LoginHeader user={user} />).toJSON()
  ).toMatchSnapshot();
});
