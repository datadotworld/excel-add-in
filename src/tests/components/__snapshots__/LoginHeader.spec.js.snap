// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login header - no user 1`] = `
<div
  className="login-header undefined"
>
  <img
    alt="data world logo"
    className="dw-logo"
    src="dw-logo.png"
  />
</div>
`;

exports[`renders login header - with user 1`] = `
<div
  className="login-header undefined"
>
  <img
    alt="data world logo"
    className="dw-logo"
    src="dw-logo.png"
  />
  <div
    className="dropdown btn-group btn-group-sm"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-sm btn-default"
      disabled={false}
      id="dropdown-user"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      type="button"
    >
      @test
       
      <span
        className="caret"
      />
    </button>
    <ul
      aria-labelledby="dropdown-user"
      className="dropdown-menu dropdown-menu-right"
      role="menu"
    >
      <li
        className=""
        role="presentation"
        style={undefined}
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          Sign out
        </a>
      </li>
    </ul>
  </div>
</div>
`;
